src/DataLogger/DataLogger.cpp

```diff
@@ -0,0 +1,248 @@
+    _curFilePtr = &curFile;
+    _numColumns = numColumns;
+    _curColumn = 0;
+    _buffer = "";
+    
+    if (_curFilePtr != nullptr)
+    {
+        DEBUG_SERIAL_LN("Creating file successful");
+        return true;
+    }
+    else
+    {
+        DEBUG_SERIAL_LN("Creating file failed.");
+        return false;
+    }
+}
+
+bool DataLogger::open(String name, int numColumns)
+{
+    _curColumn = 0;
+    _buffer = "";
+
+    // check if file exists
+    if (SD.exists(name))
+    {
+        // file already exists
+        // open the file and see how many columns are in the csv file
+
+        DEBUG_SERIAL_LN("Loading " + name);
+
+        File tempFile;
+        tempFile = SD.open(name);
+
+        // some variables to help identify how many columns in the first line
+        int commaNum = 0;
+        bool charactersPresent = false;
+        bool escaped = false;
+
+        int columnsInFile = 0;
+
+        while (tempFile.available())
+        {
+            char readChar = tempFile.read(); // char to hold a character that has been read
+
+            if (readChar == ',')
+            {
+                commaNum++;
+            }
+            else if (readChar == '\n' || readChar == '\r')
+            {
+                escaped = true;
+                break;
+            }
+            else
+            {
+                charactersPresent = true;
+            }
+        }
+
+        // determine the number of columns
+        if (commaNum > 0) {
+            columnsInFile = commaNum + 1;
+        }
+        else if (charactersPresent) {
+            columnsInFile = 1;
+        }
+        else {
+            columnsInFile = 0;
+        }
+
+        DEBUG_SERIAL_LN("Counted " + (String)columnsInFile + " columns and reached end of row.");
+
+        // check if the number of columns in the file match the number of columns specified
+        if (columnsInFile == numColumns) {
+            // file is good
+            _numColumns = numColumns;
+            _curFilePtr = &tempFile;
+            return true;
+        }
+        // check if it is an empty file
+        else if (_numColumns == 0 && escaped == false)
+        {
+            // file is empty
+            _numColumns = numColumns;
+            _curFilePtr = &tempFile;
+            return true;
+        }
+        else 
+        {
+            // file is not empty, and does not have the correct number of columns
+            DEBUG_SERIAL_LN("Incorrect column number. Creating in file copyof_" + name);
+            tempFile.close();
+
+            // create new name for file
+
+            // find extension (example: myFile.csv)
+            int extensionIndex = name.lastIndexOf('.');
+            String base = name.substring(0, extensionIndex == -1 ? name.length() : extensionIndex);
+            String extension = extensionIndex == -1 ? "" : name.substring(extensionIndex + 1);
+            base.trim();
+
+            // find current file count (example: myFile (2).csv)
```

Comment:
does this code add a space in between file name and (#)?

Reply:
Yup, it does in line 152. Do you want this to be the intended behaviour?

Reply:
Spaces in paths are a pain.  I think removing the space is a good idea

Reply:
Space has been removed